independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Priority.java

```diff
@@ -0,0 +1,20 @@
+package io.quarkus.arc;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation has the same semantics as {@link javax.annotation.Priority} except that the {@link Target} meta-annotation is
+ * not present. The main motivation is to support method and field declarations, i.e. this annotation can be used for producer
+ * methods and fields. Note that this problem is fixed in Common Annotations 2.1.
+ * <p>
+ * A priority specified by {@link AlternativePriority} and {@link javax.annotation.Priority} takes precedence.
```

Comment:
Since we quite recently had a conversation about multiple occurrences of `@Priority` for single bean, shouldn't we follow the same rules here? That is, multiple declarations of (varying) priority values should result in an error.
Quarkus will not have this problem between `AlternativePriority` and Arc `@Priority` and with jakarta artifacts, there will be jakarta `@Priority` as well.

Not to mention the mess when/is Quarkus allows priority in stereotypes. Imagine it could also allow `AlternativeStereotype` and Arc `Priority` there :-D

Reply:
That's a good point but we cannot change the current behavior where `@AlternativePriority` overrides the value from `@javax.annotation.Priority`.

> Not to mention the mess when/is Quarkus allows priority in stereotypes. Imagine it could also allow `AlternativeStereotype` and Arc `Priority` there :-D

Hm, it seems that we don't support `@AlternativePriority` on stereotypes right now. But I get your point.

Reply:
> That's a good point but we cannot change the current behavior where @AlternativePriority overrides the value from @javax.annotation.Priority.

That's good remark, I forgot about it.
I suppose the deprecation and eventual removal will take care of it.

> Hm, it seems that we don't support @AlternativePriority on stereotypes right now. But I get your point.

Will we support Arc `@Priority` on stereotypes? Or do we even support the standard one? Either way, we should define it in a similar way upcoming spec version does. Our future selves will be thankful :-D

Reply:
> Either way, we should define it in a similar way upcoming spec version does. Our future selves will be thankful :-D

I agree but that should be implemented in a separate PR... when the new spec is released ;-).

Reply:
All I am trying to make is to future-proof it so we don't allow a certain behavior now that we know will be deemed as incorrect later. But up to you, really :)

Reply:
It's not supported at the moment so I think that we're OK.

Reply:
I thought Quarkus actually supports `@Priority` on stereotypes already :-) If not, one less thing to worry about.